paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py

```diff
@@ -205,9 +206,86 @@ def run_convert():
                 "Can not convert the dataset to binary file with try limit {0}".
                 format(try_limit))
         download_concat(cache_folder, zip_path)
-        convert(zip_path, output_file)
+        convert_Imagenet_tar2bin(zip_path, output_file)
     print("\nSuccess! The binary file can be found at {0}".format(output_file))
 
 
+def convert_Imagenet_local2bin(args):
+    data_dir = args.data_dir
+    label_list_path = os.path.join(args.data_dir, args.label_list)
+    bin_file_path = os.path.join(args.data_dir, args.output_file)
+    with open(label_list_path) as flist:
+        lines = [line.strip() for line in flist]
+        num_images = len(lines)
+
+        with open(bin_file_path, "w+b") as of:
+            of.seek(0)
+            num = np.array(int(num_images)).astype('int64')
+            of.write(num.tobytes())
+            for idx, line in enumerate(lines):
+                img_path, label = line.split()
+                img_path = os.path.join(data_dir, img_path)
+                if not os.path.exists(img_path):
+                    continue
+
+                #save image(float32) to file
+                img = Image.open(img_path)
+                img = process_image(img)
+                np_img = np.array(img)
+                of.seek(SIZE_INT64 + SIZE_FLOAT32 * DATA_DIM * DATA_DIM * 3 *
+                        idx)
+                of.write(np_img.astype('float32').tobytes())
+
+                #save label(int64_t) to file
+                label_int = (int)(label)
+                np_label = np.array(label_int)
+                of.seek(SIZE_INT64 + SIZE_FLOAT32 * DATA_DIM * DATA_DIM * 3 *
+                        num_images + idx * SIZE_INT64)
+                of.write(np_label.astype('int64').tobytes())
+
+        target_size = 8 + num_images * 3 * args.data_dim * args.data_dim * 4 + num_images * 8
+        if (os.path.getsize(bin_file_path) == target_size):
+            print(
+                "Success! The user data output binary file can be found at: {0}".
+                format(bin_file_path))
+        else:
+            print("Conversion failed!")
+
+
+def main_preprocess_Imagenet(args):
+    parser = argparse.ArgumentParser(
+        description="Convert the full Imagenet val set or local data to binary file.",
+        usage=None,
+        add_help=True)
+    parser.add_argument('--local', action="store_true")
+    parser.add_argument(
+        "--data_dir",
+        default="./third_party/inference_demo/int8v2/imagenet_small",
```

Comment:
I think I would not define the default values for this path and the next filename/path options. These values would be set either in the test definition (in a cmake file) or deliberately by a user. In both cases setting the default values here can be misleading.

Reply:
You are right. I will change this.

Reply:
Done